org.adempiere.base/src/org/compiere/model/MRecordAccess.java

```diff
@@ -287,5 +289,21 @@ public String getTableName (Properties ctx)
 		}		
 		return m_tableName;
 	}	//	getTableName
+	
+	/**
+	 * 	After Save
+	 *	@param newRecord new
+	 *	@param success success
+	 *	@return success
+	 */
+	@Override
+	protected boolean afterSave(boolean newRecord, boolean success) {
+		if (!success)
+			return success;
+
+		Adempiere.getThreadPoolExecutor().submit(() -> CacheMgt.get().reset(MRole.Table_Name));

```

Comment:
could this be limited to just `CacheMgt.get().reset(MRole.Table_Name, getAD_Role_ID)` instead ?

Reply:
thanks @hengsin - implemented your suggestion